avocado/core/job.py

```diff
@@ -550,6 +553,20 @@ def run(self):
                 self.time_end = time.time()
                 self.time_elapsed = self.time_end - self.time_start
             self.__stop_job_logging()
+            self.cleanup()
```

Comment:
Note I'm not sure this is proper usage. It depends on the question asked in the main thread.